cs/src/comm/epoxy-transport/EpoxyNetworkStream.cs

```diff
@@ -369,8 +369,8 @@ static void SafeShutdownSocket(Socket socket, Logger logger)
         {
             try
             {
-                socket.Shutdown(SocketShutdown.Both);
-                socket.Close();
+                socket?.Shutdown(SocketShutdown.Both);
+                socket?.Close();
```

Comment:
We only need this for SafeShutdownSocket. EpoxyNetworkStream.Shutdown can only be called on constructed object, which have non-null Socket instances. SafeShutdownSocket is used when we're creating the socket in the first place, to then pass to the EpoxyNetworkStream ctor.

I'd rather not add summary comments to these deep internal methods in this PR.